### PR TITLE
Use titleServiceName for golang slim code generation

### PR DESCRIPTION
**Pull request by benmrosemeyer-wf**
### Story:
There is a golang problem with how a struct or union field name is built into some references while using the slim flag when the field name is all caps and contains underscores. Generally, for such field names we hit the use of [this code](https://github.com/Workiva/frugal/blob/master/compiler/generator/golang/generator.go#L2258-L2266) which short circuits for all caps field names before doing [snakeToCamel](https://github.com/Workiva/frugal/blob/ec46850af057dde90b2a78c7d9a59888cf0d54a4/compiler/generator/golang/generator.go#L2226), which is the intended behavior. With the silm flag, we go past [this gate](https://github.com/Workiva/frugal/blob/master/compiler/generator/golang/generator.go#L926-L929) and directly to a [snakeToCamel call](https://github.com/Workiva/frugal/blob/master/compiler/generator/golang/generator.go#L937) (we do that [twice](https://github.com/Workiva/frugal/blob/master/compiler/generator/golang/generator.go#L950) actually) without first checking for all caps to match intended behavior. We can fix this by calling the existing helper function [titleServiceName](https://github.com/Workiva/frugal/blob/master/compiler/generator/golang/generator.go#L2253) which matches the current logic and adds the all caps check.

### Acceptance Criteria:
- [ ] Code has been tested and results documented
- [ ] Unit tests have been updated
- [ ] Updates to documentation if necessary
- [ ] Verify and document changes to any other Messaging components
- [ ] Pull request made against the 'develop' branch, not master

### Design Notes:
No design changes. This is a bug fix.

### How To Test:
- Confirm the test additions here expose the bug https://github.com/Workiva/frugal/pull/2106
- Confirm the PR of this branch into branch above shows problem is fixed https://github.com/Workiva/frugal/pull/2107
- Confirm newly added test in this work contains the fix shown in above two steps

#### Reviewers:
@Workiva/service-platform

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by matthinrichsen-wf — APPROVED** [review]

**Comment by matthinrichsen-wf** [comment]
This approach makes sense to me - it'd be nice to have a unit test around it

**Comment by matthinrichsen-wf** [comment]
i, uh, did not expect the green "all's good" to be put on after my approval. That's terrifying.

File: compiler/generator/golang/generator.go [hunk 1]
```diff
@@ -928,13 +928,15 @@ func (g *Generator) generateWriteFieldInline(field *parser.Field) (contents stri
 		return fmt.Sprintf("\tif err := p.writeField%d(ctx, oprot); err != nil {\n\t\treturn err\n\t}\n", field.ID)
 	}
 
+	fieldName := title(field.Name)
```
Comment (benmrosemeyer-wf) [inline compiler/generator/golang/generator.go:931]: See title [here](https://github.com/Workiva/frugal/blob/master/compiler/generator/golang/generator.go#L2249-L2273)

**Review by matthinrichsen-wf — APPROVED** [review]

**Review by brettkail-wk — APPROVED** [review]

**Comment by benmrosemeyer-wf** [comment]
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**